### ISSUE TITLE
Crash report 7.2.1: ArrayIndexOutOfBoundsException in MediaGridAdapter

### ISSUE THREAD

**Issue by khaykov** [wordpress-mobile/WordPress-Android#5853]
```
Fatal Exception: java.lang.ArrayIndexOutOfBoundsException: length=181; index=-1
       at java.util.ArrayList.get(ArrayList.java:310)
       at org.wordpress.android.ui.media.MediaGridAdapter$GridViewHolder$3.onClick(MediaGridAdapter.java:293)
       at android.view.View.performClick(View.java:4508)
       at android.view.View$PerformClick.run(View.java:18675)
       at android.os.Handler.handleCallback(Handler.java:733)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:136)
       at android.app.ActivityThread.main(ActivityThread.java:5584)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:515)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1268)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1084)
       at dalvik.system.NativeStart.main(NativeStart.java)
```

**Comment by daniloercoli** [comment]
See https://github.com/wordpress-mobile/WordPress-Android/issues/5674

It seems the issue is still there, even though we fixed it in 7.2.

**Comment by daniloercoli** [comment]
The previous fix didn't make into 7.2.1 :(

In any case I saw some other parts of the code that could love some attention in regard to ArrayIndexOutOfBoundsException.

**Comment by mzorz** [comment]
should be closed via #5854
